### PR TITLE
FeedbackAssembler specs, enhancements

### PR DESCRIPTION
**Pull request by happythenewsad**
## WHAT
- The FeedbackAssembler abstract parent class had Opinion-specific logic, which would have caused grammar API to break if traffic were going to it. Fixed the bug, and added specs.
- point golang to the new LMS endpoint.

## WHY
Bugs should be fixed. 

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  yes
Have you deployed to Staging? | not yet
Self-Review: Have you done an initial self-review of the code below on Github? | yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | n/a

### CODE REVIEW

File: services/comprehension/feedback-api-main/src/endpoint/endpoint.go [hunk 1]
```diff
@@ -58,20 +58,31 @@ func GetOpinionDomain() (string) {
...
+	var maybe_domain = os.Getenv("grammar_domain")
+
+	var grammar_domain = "https://www.quill.org"
+
+	if len(maybe_domain) > 0 {
+		grammar_domain = maybe_domain
+	}
+	return grammar_domain
+}
+
```
Comment (anathomical) [inline services/comprehension/feedback-api-main/src/endpoint/endpoint.go]: Why not re-use the pattern we have for other Evidence-engine-based APIs.  Like `automl_api` just uses `lms_domain`.

File: services/QuillLMS/engines/evidence/app/controllers/evidence/grammar_controller.rb [hunk 2]
```diff
@@ -2,8 +2,8 @@ module Evidence
   class GrammarController < ApiController
 
     def fetch
-      oapi_client = Grammar::Client.new(entry: params['entry'], prompt_text: params['prompt_text'])
-      render json: Grammar::FeedbackAssembler.run(oapi_client.post)
+      grammar_client = Grammar::Client.new(entry: params['entry'], prompt_text: params['prompt_text'])
+      render json: Grammar::FeedbackAssembler.run(grammar_client.post)
     end
   end
 end
```
Comment (anathomical) [inline services/QuillLMS/engines/evidence/app/controllers/evidence/grammar_controller.rb]: What if we just put this (and the Opinion API Controller) into the `FeedbackController`.  There are already endpoints there for different feedback types.  Consolidation of these single-action controllers is probably easier to keep track of.

File: services/QuillLMS/engines/evidence/lib/evidence/evidence/feedback_assembler.rb [hunk 3]
```diff
@@ -33,6 +33,9 @@ def self.default_payload
         'rule_uid': ''
       }  
     end
-  end
-    
+
+    def self.error_name 
+      'abstract_error'
```
Comment (anathomical) [inline services/QuillLMS/engines/evidence/lib/evidence/evidence/feedback_assembler.rb]: Maybe just `raise NoMethodError` or something here in the abstract class?  That'll actually cause errors if someone fails to do a sub-class implementation which I think is the behavior we want?

**Review by anathomical — CHANGES_REQUESTED** [review]
I like the implementation in general, and appreciate the tests a ton.  A couple of thoughts on consolidation that might make things simpler?  Some of these probably should have been recommended back when the original OAPI changes were merged.

**Review by cissyxyu — APPROVED** [review]

**Comment by happythenewsad** [comment]
@anathomical @cissyxyu Thanks for reviewing. This PR is blocked on https://www.notion.so/quill/blocked-Set-up-staging-Spell-model-servers-562b17d99b0b4266930c0fb227765420

I will address your changes once the PR becomes unblocked.

**Review by cissyxyu — APPROVED** [review]

**Review by anathomical — APPROVED** [review]

**Pull request merged**